app/services/filter_application_choices_for_providers.rb

```diff
@@ -60,7 +60,9 @@ def accredited_provider(application_choices, accredited_providers)
     def provider_location(application_choices, provider_location)
       return application_choices if provider_location.blank?
 
-      application_choices.where(site: { id: provider_location })
+      name, code = provider_location[0].split('|')
+
+      application_choices.where(site: { name: name }).where(site: { code: code })
```

Comment:
fwiw, this could be `application_choices.where(site: { name: name, code: code })`